imports/packages/callhistory/callhistory.tsx

```diff
@@ -0,0 +1,150 @@
+import { registerPlugin } from '@capacitor/core';
+import { DeepClient } from "@deep-foundation/deeplinks/imports/client";
+interface IPermissionStatus {
+  granted: boolean;
+}
+
+export interface ICallHistory {
+  phoneNumber: string;
+  type: string;
+  date: string;
+  duration: string;
+}
+
+interface ICallHistoryPlugin {
+  getPermissions(): Promise<IPermissionStatus>;
+  getCallHistory(): Promise<{ call_log: ICallHistory[] }>;
+
+}
+
+export const CallLog = registerPlugin<ICallHistoryPlugin>('CallHistory');
+
+
+export async function createAllCallHistory({ deep, deviceLinkId }: { deep: DeepClient, deviceLinkId: any }) {
+  await CallLog.getPermissions();
+
+  const callInfoType = await deep.id(
+    "@deep-foundation/callhistory",
+    "callInfo"
+  );
+  const phoneNumberType = await deep.id(
+    "@deep-foundation/callhistory",
+    "phoneNumber"
+  );
+  const typeType = await deep.id(
+    "@deep-foundation/callhistory",
+    "type"
+  );
+  const dateType = await deep.id(
+    "@deep-foundation/callhistory",
+    "date"
+  );
+  const durationType = await deep.id(
+    "@deep-foundation/callhistory",
+    "duration"
+  );
+
+  const ml = new ManagerLinks(deep);
+  const res = await CallLog.getCallHistory();
+  for (const call_log of res.call_log) {
+    const callInfoLink = await ml.createLink({
+      containName: "",
+      contain_from_id: deviceLinkId,
+      type_id: callInfoType,
+    });
+    const phoneNumber = await ml.createLink({
+      containName: "phoneNumber",
+      value: call_log.phoneNumber,
+      contain_from_id: callInfoLink,
+      type_id: phoneNumberType,
+    });
+    const type = await ml.createLink({
+      containName: "type",
+      value: call_log.type,
+      contain_from_id: callInfoLink,
+      type_id: typeType,
+    });
+    const date = await ml.createLink({
+      containName: "date",
+      value: call_log.date,
+      contain_from_id: callInfoLink,
+      type_id: dateType,
+    });
+    const duration = await ml.createLink({
+      containName: "duration",
+      value: call_log.duration,
+      contain_from_id: callInfoLink,
+      type_id: durationType,
+    });
+    break;
+  }
+}
+
+export async function createCallHistoryPackage({ deep, deviceLinkId }: { deep: DeepClient, deviceLinkId: any }) {
+  const typeTypeLinkId = await deep.id("@deep-foundation/core", "Type");
+  const packageTypeLinkId = await deep.id("@deep-foundation/core", "Package");
+
+  const ml = new ManagerLinks(deep);
+
+  const callhistoryPackage = await ml.createLink({
```

Comment:
Reduce amount of deep network requests 

_This comment applies to all the places working like these lines of code_